src/index.ts

```diff
@@ -275,8 +272,22 @@ export function sendDistributionMetric(name: string, value: number, ...tags: str
 
   if (currentMetricsListener !== undefined) {
     currentMetricsListener.sendDistributionMetric(name, value, false, ...tags);
-  } else {
-    logError("handler not initialized");
+    return;
+  }
+
+  _metricsQueue.push({ name, value, tags });
+}
+
+function sendQueueMetrics(listener: MetricsListener) {
+  while (_metricsQueue.length > 0) {
+    const metric = _metricsQueue.shift()!; // This will always exist.
```

Comment:
two question on the `shift()` part:
1. i think it's O(n) time complex. why don't we just use an index to loop.
2. i assume because we also want to clean up the queue. this makes me wonder if we could run into some corner cases such as timeouts and results in a dangling queue?

Reply:
Ahhh, you are right, this is definitely slow. 

I might either:
1. Use the `pop` method since it's `O(1)`, but metrics will arrive in a LIFO order, ending in a `O(n)`.
2. Use the previous approach, but reverse before sending data, instead of being `O(n ** 2)`, it would be at most `O(2n)` => `O(n)`, WDYT?

> i assume because we also want to clean up the queue. this makes me wonder if we could run into some corner cases such as timeouts and results in a dangling queue?
Not worried about timeouts, since this operation should be really fast, even with a thousand metrics.
On dangling queues, this shouldn't be possible, since we clean it on every flush. If there's no flush, then once the queue is full, it will be reset. WDYT?

Reply:
Updated and use the second strategy as per our conversation!